packages/perseus/src/widgets/matrix/matrix.tsx

```diff
@@ -100,10 +102,31 @@ export function getMatrixSize(matrix: ReadonlyArray<ReadonlyArray<number>>) {
 }
 
 type ExternalProps = WidgetProps<
-    PerseusMatrixWidgetOptions,
+    {
```

Comment:
It's needed because of the way TypeScript deals with React default props.

The `Props` type argument to `React.Component` represents the types of the props _inside_ the component, after `defaultProps` have been applied. However, we're using the `*WidgetOptions` types to represent the props that are _passed to_ the widget component. It's not correct to use `*WidgetOptions` as the Props type argument if the component has default props.

The specific change that motivated this was that I made `prefix` and `suffix` optional on `PerseusMatrixWidgetOptions`. I did that because `prefix` and `suffix` aren't always present in the JSON. But inside the component, they will always be strings, because they have `defaultProps`.